arelle/Cntlr.py

```diff
@@ -153,10 +154,13 @@ def __init__(
         }
         self.hasWin32gui = False
         self.hasGui = hasGui
-        self.hasFileSystem = True # no file system on Google App Engine servers
-        self.isGAE = False
-        self.isCGI = False
-        self.systemWordSize = int(round(math.log(sys.maxsize, 2)) + 1) # e.g., 32 or 64
+        self.hasFileSystem = hasFileSystem() # no file system on Google App Engine servers
+        self.isGAE = isGAE()
+        self.isCGI = isCGI()
```

Comment:
`self.hasWebServer` is still set below using a try except block. Reuse the function from SystemInfo.